### PR TITLE
feat: publish styles as independant package

### PR DESCRIPTION
**Pull request by lee-chase**
Contributes to #3716

Created a new package by copying the ibm-products package.json and cutting it down to make  `ibm-products-styles`.

It copies the styles from `packages/ibm-products/src` and builds a package from them. 

Note: That creating a copy of the source is in part based on the assumption that something in the package would need to change in order to bump the version. This should be revisited as a final strategy, it would probably be better to have both @carbon/ibm-products and @carbon/ibm-products-styles have the same version with changes to styles or components causing the other to update it's version.

#### What did you change?

New package

#### How did you test and verify your work?

Test build and `npm pack` in the `packages/ibm-products-styles` folder to see what would be published.

### CODE REVIEW

File: .github/workflows/release.yml
Comment (elycheea) [inline .github/workflows/release.yml:1]: Let’s put this as a prerelease workflow instead?

Comment (lee-chase) [reply .github/workflows/release.yml:1]: @elycheea done sort of

**Comment by matthewgallo** [comment]
I think the build script is failing because `InlineTip` and `SteppedAnimatedMedia` components were merged this morning, I'm guessing their styles need to be moved now.

File: .github/workflows/release-scheduled.yml
Comment (elycheea) [inline .github/workflows/release-scheduled.yml:12]:
```suggestion
      - uses: ./release.yml
```

File: .github/workflows/release-preminor.yml
Comment (elycheea) [inline .github/workflows/release-preminor.yml:12]:
```suggestion
      - uses: ./release.yml
```

File: packages/ibm-products-styles/jest.config.js
Comment (elycheea) [inline packages/ibm-products-styles/jest.config.js:2]:
```suggestion
/**
 * Copyright IBM Corp. 2023, 2023
```

File: packages/ibm-products-styles/src/__tests__/test-helper.js
Comment (elycheea) [inline packages/ibm-products-styles/src/__tests__/test-helper.js:2]:
```suggestion
//
// Copyright IBM Corp. 2023, 2023
```

**Review by elycheea — COMMENTED** [review]
Minor changes in comments.

As we discussed previously, we’ll probably want to update our example component and our own contribution docs with these changes too. We can make those in a new issue. I know Jeff and Paul used `generate` pretty religiously for the n2p components so we’ll need to ensure those SCSS files get moved to the styles package correctly.

**Comment by elycheea** [comment]
`yarn generate` works great — might not be clear to someone new that styles are located in a different package (since it use uses the same `src/components/` path regardless) although they can likely figure it out. Can open an issue to update our getting started guides.

image

**Review by elycheea — DISMISSED** [review]

**Comment by elycheea** [comment]
Approved but wait until after next release to merge so we have time to resolve any build-related issues.

**Review by elycheea — APPROVED** [review]
Re-approving. ✔️

**Pull request merged**
